### PR TITLE
Time Server

### PR DESCRIPTION
**Pull request by AydinHassan**
TODO 
- [x] Problem
- [x] Tests

### CODE REVIEW

**Comment by mikeymike** [comment]
Travis failed... got the same error locally trying too after `composer install` but works after `composer update` :+1:

**Comment by mikeymike** [comment]
Also getting exception when running verify against the solution. It's the unsafe to use system timezone so potential simple fix. Don't think we can rely on the user to have configured PHP to prevent this.

**Comment by AydinHassan** [comment]
@mikeymike I think I'm gonna tackle this a different way. Basically any date exercise we would need to tell the user to do this, which is annoying. I'm thinking maybe we can just do it automatically, like inject code. So I'm going to investigate a way maybe exercises can prepend code or wrap code in submissions.

**Comment by AydinHassan** [comment]
Hhmm that test failure is annoying. Maybe that number isn't consistent across versions, i'll have to do a regex match

**Comment by AydinHassan** [comment]
:rage4:  GOD DAMN WINDOWS

File: src/TcpSocketFactory.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Class SocketFactory
+ * @package PhpSchool\LearnYouPhp
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class TcpSocketFactory
+{
+    /**
+     * @param string $ip
+     * @param int $port
+     * @return Cliente
```
Comment (AydinHassan) [inline src/TcpSocketFactory.php]: Typo

**Pull request merged**